### DISCUSSION TITLE
MVP

### DISCUSSION THREAD

**Discussion by brillout** [magne4000/universal-middleware#6] [General]
I wonder what an MVP could look like.

If we skip support for HTML Streaming for now, maybe there is a way to have an MVP that is fairly simple (e.g. that doesn't use any mutation observing) and that already works for Bati. I'm guessing for Bati for now we only need two universal handlers (Vike + Telefunc) and one universal middleware (Authentication).

(Although HTML Streaming is quite essential, especially for React, but maybe there is an easy way to support streams: in principle it should be enough if Universal Handler merely passes the stream around and doesn't need to manipulate the stream itself.)

For example, we don't need [order magic numbers](https://github.com/magne4000/universal-handler/discussions/4) for now, and I guess we don't need [universal Express.js middlewares](https://github.com/magne4000/universal-handler/discussions/2#discussioncomment-9461076) to support Bati's use case.

Maybe the MVP could even have zero-dependencies. Maybe we can build on top of a lean core.

(It's neat to have discussions and experimentations to minimize blind spots and minimize slowing-down surprises down the line.)

**Comment by magne4000** [comment]
> I'm guessing for Bati for now we only need two universal handlers (Vike + Telefunc) and one universal middleware (Authentication).

Exactly, and I think I can have streaming support even in a MVP (it's a must have, and is architecturally structuring).

> Maybe we can build on top of a lean core.

That also was my though on this whole endeavor. If ([when](https://github.com/magne4000/universal-handler/discussions/2#discussioncomment-9470560)) we ever need ponyfills, we should add them in dedicated repos only to limit bloat and perf impact. And I don't think we'll need utils libs, at least not for now.

**Comment by magne4000** [comment]
https://github.com/batijs/bati/pull/259

**Comment by magne4000** [comment]
MVP done in https://github.com/batijs/bati/pull/259

I'll extract the necessary information in dedicated discussions.
